ordered_set.py

```diff
@@ -335,15 +337,15 @@ def union(self, *sets: SetLike[T]) -> "OrderedSet[T]":
         Each items order is defined by its first appearance.
 
         Example:
-            >>> oset = OrderedSet.union(OrderedSet([3, 1, 4, 1, 5]), [1, 3], [2, 0])
+            >>> oset = OrderedSet([3, 1, 4, 1, 5]).union([1, 3], [2, 0])
             >>> print(oset)
             OrderedSet([3, 1, 4, 5, 2, 0])
             >>> oset.union([8, 9])
             OrderedSet([3, 1, 4, 5, 2, 0, 8, 9])
             >>> oset | {10}
             OrderedSet([3, 1, 4, 5, 2, 0, 10])
         """
-        cls = self.__class__ if isinstance(self, OrderedSet) else OrderedSet
+        cls = self.__class__
```

Comment:
I'm not sure where this idiom came from. I think the intent is to gracefully handle subclasses of `OrderedSet` so that you will call the constructor for `MyCustomOrderedSet` rather than always `OrderedSet`? 

If that's the case, using `self.__class__` will do the trick. It will always be an instance of `OrderedSet` so the else class was never triggering.